src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/PileupDetectionArgumentCollection.java

```diff
@@ -0,0 +1,92 @@
+package org.broadinstitute.hellbender.tools.walkers.haplotypecaller;
+
+import org.broadinstitute.barclay.argparser.Advanced;
+import org.broadinstitute.barclay.argparser.Argument;
+
+/**
+ * Set of arguments for configuring the pileup detection code
+ */
+public final class PileupDetectionArgumentCollection {
+
+    public static final String PILEUP_DETECTION_LONG_NAME = "pileup-detection";
+    public static final String PILEUP_DETECTION_ENABLE_INDELS = "pileup-detection-enable-indel-detection";
+    public static final String PILEUP_DETECTION_SNP_THRESHOLD = "pileup-detection-enable-snp-alt-threshold";
+    public static final String PILEUP_DETECTION_ABSOLUTE_ALT_DEPTH = "pileup-detection-absolute-alt-depth";
+    public static final String PILEUP_DETECTION_INDEL_THRESHOLD = "pileup-detection-enable-indel-alt-threshold";
+    public static final String PILEUP_DETECTION_FILTER_COVERAGE_LONG_NAME = "pileup-detection-filter-coverage-threshold";
+
+    //TODO we currently don't see the same threshold from active region determination...
+    public static final String PILEUP_DETECTION_ACETIVE_REGION_LOD_THRESHOLD_LONG_NAME = "pileup-detection-active-region-lod-threshold";
+
+
+    // Arguments related to DRAGEN heuristics related to "read badness" intended to filter out false positives from the pileup deteciotn code
+    public static final String PILEUP_DETECTION_BAD_READ_RATIO_LONG_NAME = "pileup-detection-bad-read-tolerance";
+    public static final String PILEUP_DETECTION_PROPER_PAIR_READ_BADNESS_LONG_NAME = "pileup-detection-proper-pair-read-badness";
+    public static final String PILEUP_DETECTION_EDIT_DISTANCE_BADNESS_LONG_NAME = "pileup-detection-edit-distance-read-badness-threshold";
+    public static final String PILEUP_DETECTION_CHIMERIC_READ_BADNESS_LONG_NAME = "pileup-detection-chimeric-read-badness";
+    //TODO these need to be implemented with some input from Illumina
+    public static final String PILEUP_DETECTION_TLEN_MEAN_LONG_NAME = "pileup-detection-template-mean-badness-threshold";
+    public static final String PILEUP_DETECTION_TLEN_STD_LONG_NAME = "pileup-detection-template-std-badness-threshold";
+
+    // Argumetns related to filtering the list at the assembly graph level
+    public static final String PILEUP_DETECTION_INDEL_SNP_BLOCKING_RANGE = "pileup-detection-snp-adjacent-to-assembled-indel-range";
+
+    /**
+     * Enables pileup-based haplotype creation and variant detection
+     *
+     * NOTE: --pileup-detection is a beta feature. Use this mode at your own risk.
+     */
+    @Argument(fullName= PILEUP_DETECTION_LONG_NAME, doc = "If enabled, the variant caller will create pileup-based haplotypes in addition to the assembly-based haplotype generation.", optional = true)
```

Comment:
do we have a `@beta` annotation? or maybe use `@advanced`?

Reply:
It looks like the only options for arguments in GATK are `@Advanced` and `@Hidden`. Perhaps its better to hide all of these then? Or perhaps hide all of the parameters for this mode and leave the switch as advanced? I'll go with that unless you have opinions. @yfarjoun